pkg/test/framework/components/echo/check/checkers.go

```diff
@@ -363,10 +365,17 @@ func URL(expected string) echo.Checker {
 
 // ReachedTargetClusters is similar to ReachedClusters, except that the set of expected clusters is
 // retrieved from the Target of the request.
-func ReachedTargetClusters(allClusters cluster.Clusters) echo.Checker {
+func ReachedTargetClusters(t framework.TestContext) echo.Checker {
+	dnsCaptureEnabled := istio.GetOrFail(t, t).MeshConfigOrFail(t).DefaultConfig.ProxyMetadata["ISTIO_META_DNS_CAPTURE"] == "true"
```

Comment:
this function is probably not super fast... is it something we can call in a Checker efficiently?

Reply:
Oh we are caching it. Hopefully accurate, given other things can change it...

We could have a function that always fetches fresh, and use a cached variant here, if its an issue.

Reply:
Just to be clear, are you suggesting that we have separate checkers for cached/fresh values? I'm not sure we want the tests to care.

WRT caching ... The way the caching works is that it gets updated each time the test environment changes meshconfig. I think that should be sufficient WRT correctness, no?

Reply:
No, checker should always cache certainly or performance will be terrible. But there are other usages to get MeshConfig, and they may not play well with cached result.

This is all hypothetical.. if all tests pass we can just keep it as is and if needed in the future add a way to get an uncached version.

> WRT caching ... The way the caching works is that it gets updated each time the test environment changes meshconfig. I think that should be sufficient WRT correctness, no?

Possibly, but there is nothing enforcing that it isn't changed by other things (Istio operator, for example)